pandas/core/index.py

```diff
@@ -2226,7 +2228,9 @@ def drop(self, labels):
         indexer = self.get_indexer(labels)
         mask = indexer == -1
         if mask.any():
-            raise ValueError('labels %s not contained in axis' % labels[mask])
```

Comment:
hmm, shouldn't this raise a ``KeyError``? @jorisvandenbossche @shoyer to be consistent with say ``.loc`` and other indexers? @hayd